### PR TITLE
MCKIN-11429 Configure tox with edx-notifications

### PR DESCRIPTION
**Pull request by moeez96**
Ticket: https://edx-wiki.atlassian.net/browse/MCKIN-11429

Configure Tox with edx-notifications to support test suite on both python 2.7+Django1.11 and python 3.5+Django (to be added later once edx-notifications is updated to that python version.)

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @moeez96! I've created [OSPR-3815](https://openedx.atlassian.net/browse/OSPR-3815) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).



We can't start reviewing your pull request until you've submitted a     [signed contributor agreement](https://open.edx.org/wp-content/uploads/2019/01/individual-contributor-agreement.pdf)     or indicated your institutional affiliation. Please see the [CONTRIBUTING](https://github.com/edx/edx-platform/blob/master/CONTRIBUTING.rst)     file for more information.

**Comment by natabene** [comment]
@ihtram Please review soon.

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,15 @@
+[tox]
+envlist = py27-django111
```
Comment (ihtram) [inline tox.ini]: @moeez96 We need to configure tox for both py27 and py35. Please configure tox for py35 as well.

Comment (ihtram) [reply tox.ini]: also add allow_failures in travis config for py35 to ignore py35 failures for now.

File: .travis.yml [hunk 2]
```diff
@@ -7,10 +7,12 @@ addons:
 
 env:
   - DJANGO_SETTINGS_MODULE=settings
+sudo: required
```
Comment (ihtram) [inline .travis.yml]: @moeez96 Why do we need root privilege here?

File: tox.ini (hunk 1 above)
Comment (moeez96) [reply tox.ini]: @ihtram py35 is not a part of travis env hence adding allow_failures in travis for a tox config doesnt seems well. How about we keep py27 for this pr in tox and add py35 in tox with the next PR which actually contains all py35 tests. What do you think?

File: .travis.yml (hunk 2 above)
Comment (moeez96) [reply .travis.yml]: Removed

File: tox.ini (hunk 1 above)
Comment (ihtram) [reply tox.ini]: we should configure py35 for travis. Adding tox for only py27 is meaningless as the purpose of tox is to run multiple environments. After configuring tox for both versions we can then start fixing broken tests.

**Comment by moeez96** [comment]
@ihtram Please review

**Comment by ihtram** [comment]
@moeez96 We still need to configure py35 with travis. So basically this should run tests on both py27 and py35. py35 tests will fail but you can add configurations to allow_failures for py35. Once this is done you need to rebase the other PR(https://github.com/edx/edx-notifications/pull/198) where unit tests have been fixed but those tests are running only for py27. 
So basically this should add support to run tests with both versions of python something like this:
https://github.com/edx/ecommerce/blob/master/.travis.yml

**Comment by moeez96** [comment]
@ihtram py35 addedd to travis, Rebasing to PR #198 now.

**Comment by moeez96** [comment]
@ihtram PR rebased, Failing tests will be solved in separate PR
Cc: @nasirhjafri

**Comment by moeez96** [comment]
@ihtram please have a look

**Comment by edx-webhook** [comment]
Although this pull request is already merged, I've created [OSPR-5418](https://openedx.atlassian.net/browse/OSPR-5418) so that we can track it in Jira.

There is nothing you have to do. No action is needed from your side.  Thanks again for your contribution.

**Pull request merged**